src/NAnt.Core/Tasks/LoopTask.cs

```diff
@@ -429,9 +429,15 @@ public enum LoopTrim {
                         break;
                 }
             } finally {
-                // Restore all of the old property values
-                for (int nIndex = 0; nIndex < oldPropVals.Length; nIndex++) {
-                    Properties[_props[nIndex]] = oldPropVals[nIndex];
+                // Restore all of the old property values. If any of the old properties
+                // contained a null value, remove that property from the dictionary.

```

Comment:
Yeah, I think you are correct. That particular note was from a slightly different approach before I changed it to this. It should reflect correctly. I'll make that change today.